cocotb/drivers/amba.py

```diff
@@ -198,6 +204,8 @@ def read(self, address, sync=True):
                 break
             yield RisingEdge(self.clock)
 
+        yield RisingEdge(self.clock)
```

Comment:
This doesn't look like it belongs in this patch

Reply:
@eric-wieser, I copied the content of the `write` function into the `read` function for the sake of uniformity. Now both write and read functions end one rising edge after the slave has asserted `BVALID` or `RVALID`, which is the effective end of transaction.

Reply:
That may be, but I've seen a lot of code that depends on this edge _not_ being there.

At any rate, this is not part of the change to use a lock - its best to put unrelated changes like this line in a separate PR, so that they can be reviewed independently.

Reply:
Okay, I removed that line.

Does it make sense to uniformize both functions? If a lot of code depends on this edge not being there, should I do it the other way - remove that edge in the write function as well?

Reply:
Well, it's likely that code depends on the edge not being there in the other function too - best to leave it for now, and maybe make a brand new `cocotb_ext.axilite` module in the somewhat distant future with all these things fixed.